src/client/datascience/jupyter/jupyterExecution.ts

```diff
@@ -233,30 +225,17 @@ export class JupyterExecutionBase implements IJupyterExecution {
                             traceError('Failed to connect to server', ex);
                             if (ex instanceof JupyterSessionStartError && isLocalConnection && allowUI) {
                                 // Keep retrying, until it works or user cancels.
-                                // Sometimes if a bad kernel is selected, starting a session can fail.
-                                // In such cases we need to let the user know about this and prompt them to select another kernel.
-                                const message = localize.DataScience.sessionStartFailedWithKernel().format(
-                                    getDisplayNameOrNameOfKernelConnection(launchInfo.kernelConnectionMetadata),
-                                    Commands.ViewJupyterOutput
+                                const kernelInterpreter = await this.kernelSelector.askForLocalKernel(
```

Comment:
Should we just exit out of here and throw an error instead.
In Native Notebooks, this isn't going to work, the kernel tha'ts selected & displayed will be different from whats going to be used. I understand this is how things have worked in webview/interactive.

Ideally at least in native notebooks, we should ask the user to select a different kernel.

Or how about we add a telemetry point to see how often this happens.
Because if this happens in NativeNotebooks, we'll always be saving the wrong kernel in the notebook, i suspect it will be the case even with webviews.
I.e. in notebooks they open, its wrong & gets changed, but old kernel is still selected & they save. They open again, its it using the incorrect kernel.

Reply:
Yeah I'd rather not change that logic yet. I'll add the telemetry point.